udata/core/dataservices/search.py

```diff
@@ -0,0 +1,121 @@
+        if args.get("q"):
+            # Following code splits the 'q' argument by spaces to surround
+            # every word in it with quotes before rebuild it.
+            # This allows the search_text method to tokenise with an AND
+            # between tokens whereas an OR is used without it.
+            phrase_query = " ".join([f'"{elem}"' for elem in args["q"].split(" ")])
+            dataservices = dataservices.search_text(phrase_query)
+        if args.get("tag"):
+            dataservices = dataservices.filter(tags=args["tag"])
+        if args.get("organization"):
+            if not ObjectId.is_valid(args["organization"]):
+                api.abort(400, "Organization arg must be an identifier")
+            dataservices = dataservices.filter(organization=args["organization"])
+        if "is_restricted" in args:
+            dataservices = dataservices.filter(is_restricted=boolean(args["is_restricted"]))
+        return dataservices
+
+
+@register
+class DataserviceSearch(ModelSearchAdapter):
+    model = Dataservice
+    search_url = "dataservices/"
+
+    sorts = {
+        "created": "created_at",
+    }
+
+    filters = {
+        "tag": Filter(),
+        "organization": ModelTermsFilter(model=Organization),
+        "archived": BoolFilter(),
+    }
+
+    @classmethod
+    def is_indexable(cls, dataservice: Dataservice) -> bool:
+        return dataservice.deleted_at is None and not dataservice.private
+
+    @classmethod
+    def mongo_search(cls, args):
+        dataservices = Dataservice.objects(deleted_at=None, private__ne=True)
+        dataservices = DataserviceApiParser.parse_filters(dataservices, args)
+
+        sort = (
+            cls.parse_sort(args["sort"])
+            or ("$text_score" if args["q"] else None)
+            or DEFAULT_SORTING
+        )
+        offset = (args["page"] - 1) * args["page_size"]
+        return dataservices.order_by(sort).skip(offset).limit(
+            args["page_size"]
+        ), dataservices.count()
+
+    @classmethod
+    def serialize(cls, dataservice: Dataservice) -> dict:
+        organization = None
+        owner = None
+        if dataservice.organization:
+            org = Organization.objects(id=dataservice.organization.id).first()
+            organization = {
+                "id": str(org.id),
+                "name": org.name,
+                "public_service": 1 if org.public_service else 0,
+                "followers": org.metrics.get("followers", 0),
+            }
+        elif dataservice.owner:
+            owner = User.objects(id=dataservice.owner.id).first()
+        extras = {}
+        for key, value in dataservice.extras.items():
+            extras[key] = to_iso_datetime(value) if isinstance(value, datetime.datetime) else value
+
+        return {
+            "id": str(dataservice.id),
+            "title": dataservice.title,
+            "description": dataservice.description,
+            "base_api_url	": dataservice.base_api_url,
```

Comment:
is that a typo, having the extraneous spaces after `has_api_url`?

Reply:
:+1: fixed in https://github.com/opendatateam/udata/pull/3171/commits/c275f844d0ba32f142348a90b44182ef0986d7b9